.github/workflows/tests.yml

```diff
@@ -273,22 +273,40 @@ jobs:
         # does not yet support memory64.
         run: cargo test --no-run -Z build-std=std,panic_abort --target=wasm64-unknown-unknown --features=js
 
-  wasi-tests:
-    name: WASI Test
+  wasip1-tests:
+    name: WASI Preview 1 Test
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@stable
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          targets: wasm32-wasi
+          targets: wasm32-wasip1
+      - name: Install precompiled wasmtime
+        run: |
+          VERSION=v24.0.0
+          URL=https://github.com/bytecodealliance/wasmtime/releases/download/${VERSION}/wasmtime-${VERSION}-x86_64-linux.tar.xz
+          wget -O - $URL | tar -xJ --strip-components=1 -C ~/.cargo/bin
+          wasmtime --version
+      - uses: Swatinem/rust-cache@v2
+      - run: cargo test --no-default-features --target wasm32-wasip1
+
+  wasip2-tests:
+    name: WASI Preview 2 Test
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
+        with:
+          targets: wasm32-wasip2
+          components: rust-src
       - name: Install precompiled wasmtime
         run: |
-          VERSION=v2.0.0
+          VERSION=v24.0.0
           URL=https://github.com/bytecodealliance/wasmtime/releases/download/${VERSION}/wasmtime-${VERSION}-x86_64-linux.tar.xz
           wget -O - $URL | tar -xJ --strip-components=1 -C ~/.cargo/bin
           wasmtime --version
       - uses: Swatinem/rust-cache@v2
-      - run: cargo test --target wasm32-wasi
+      - run: cargo test --no-default-features --target wasm32-wasip2 -Z build-std=core,alloc
```

Comment:
@LikeLakers2
Can you help with fixing the wasmtime-based CI jobs? The current config results in the following errors: [p1](https://github.com/rust-random/getrandom/actions/runs/10720800811/job/29727858197?pr=499), [p2](https://github.com/rust-random/getrandom/actions/runs/10720800811/job/29727858835?pr=499).

Reply:
I worked around this issue by using `--release` flag for wasip2 tests.